File: lab4/backend/products/soa-products-rest-adapter/src/main/kotlin/com/soa/products/rest/adapter/wsdl/ColorTo.java

package com.soa.products.rest.adapter.wsdl;

import jakarta.xml.bind.annotation.XmlEnum;
import jakarta.xml.bind.annotation.XmlType;


/**
 * 
 * 
 * <p>Java class for colorTo</p>.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.</p>
 * <pre>{@code
 * <simpleType name="colorTo">
 *   <restriction base="{http://www.w3.org/2001/XMLSchema}string">
 *     <enumeration value="YELLOW"/>
 *     <enumeration value="WHITE"/>
 *     <enumeration value="BROWN"/>
 *     <enumeration value="GREEN"/>
 *     <enumeration value="RED"/>
 *     <enumeration value="BLACK"/>
 *     <enumeration value="ORANGE"/>
 *   </restriction>
 * </simpleType>
 * }</pre>
 * 
 */
@XmlType(name = "colorTo")
@XmlEnum
public enum ColorTo {

    YELLOW,
    WHITE,
    BROWN,
    GREEN,
    RED,
    BLACK,
    ORANGE;

    public String value() {
        return name();
    }

    public static ColorTo fromValue(String v) {
        return valueOf(v);
    }

}

File: lab4/backend/products/soa-products/src/main/kotlin/com/soa/products/dto/ColorTo.kt
package com.soa.products.dto

enum class ColorTo(val value: String) {
    YELLOW("YELLOW"),
    WHITE("WHITE"),
    BROWN("BROWN"),
    GREEN("GREEN"),
    RED("RED"),
    BLACK("BLACK"),
    ORANGE("ORANGE");

    companion object {
        fun forValue(value: String): ColorTo {
            return values().firstOrNull { it.value == value }
                ?: throw IllegalArgumentException("No enum constant with value $value")
        }
    }
}
